rules.d/60-block.rules

```diff
@@ -11,3 +11,6 @@ ACTION=="change", SUBSYSTEM=="scsi", ENV{DEVTYPE}=="scsi_device", TEST=="block",
 ACTION!="remove", SUBSYSTEM=="block", \
   KERNEL=="loop*|mmcblk*[0-9]|msblk*[0-9]|mspblk*[0-9]|nvme*|sd*|vd*|xvd*|bcache*|cciss*|dasd*|ubd*|ubi*|scm*|pmem*|nbd*|zd*", \
   OPTIONS+="watch"
+
+# apply known iocost solutions
+ACTION=="add", SUBSYSTEM=="block", RUN+="/usr/lib/systemd/systemd-iocost apply $sys$devpath"
```

Comment:
Maybe, it is better to use a template service `systemd-iocost@.service`??

Reply:
I think this makes sense, thanks for the suggestion! Will do this change after I push the other fixes.

By the way, do you have any recommendations on who would be a good reviewer to take a look at the proposed functionality? I am planning to look up people who work on hwdb and devices / cgroups handling and ask them.

Thanks for the review!

Reply:
Maybe, @poettering, @bluca, @anitazha, @htejun.

Reply:
this is shortlived stuff, no? not long running? if it just writes something to cgroupfs or so and immediately exits, then a service is not necessary, just make it something invoked by RUN+.

if this is expected to be run on the majority of systems for a multitude of devices it should even be made a builtin, i.e. the way kmod or uaccess are...

Reply:
It is quite simple indeed if we go with a single set of model/qos for each device. I was wondering if it made sense to just use echo to set the model and qos cgroupfs, probably makes sense to use the binary to use libsystemd and rely on its error handling and logging, though? In any case, for now I went with the very simple binary (that we can extend further later with more features, maybe even to download the database?) using RUN.